CalendarDisplay/ViewModels/CalendarViewModel.cs

```diff
@@ -96,21 +96,22 @@ public CalendarGrid CreateCalendar(DateTime date, List<EventViewModel> eventView
 
             // We start with some empty days to represent the days at the end of the previous month
             var lastMonthsDays = Enumerable.Range(0, startingDayOfMonthDayOfWeek)
-                .Select(day => new CalendarDay(startingDayOfMonth.AddDays(-(1 + day)), new()))
+                .Select(day => new CalendarDay(startingDayOfMonth.AddDays(-(1 + day))))
                 .Reverse()
                 .ToList();
 
             // Start looping though each day of the month to fetch any events
             for (int day = 1; day <= daysInMonth; day++)
             {
-                var daysEvents = eventViewModels.Where(e => e.Start.Day == day).ToList();
```

Comment:
The bug is here, where it matches up events based on only the dates day value, and does not take into account the year or month. Because of this, the event was being duplicated when it shouldn't of been.